### PR TITLE
Rearrange String.prototype.indexOf, lastIndexOf and Replace helpers

### PR DESCRIPTION
**Pull request by lvidacs**
Fixes issue #515

JerryScript-DCO-1.0-Signed-off-by: Laszlo Vidacs lvidacs.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by ruben-ayrapetyan** [comment]
@lvidacs, please, fix precommit and style issues (like missing comments to function arguments).

File: tests/jerry/string-prototype-indexof.js [hunk 1]
```diff
@@ -85,33 +85,17 @@ assert("Hello world, welcome to the universe.".indexOf("welcome", undefined_var)
 // check booleans
 assert("true".indexOf(true, false) === 0);
 
-// check this is undefined
-try {
-  String.prototype.indexOf.call(undefined);
```
Comment (galpeter) [inline tests/jerry/string-prototype-indexof.js]: Why should we remove these test cases?

**Comment by egavrin** [comment]
good to me

**Comment by tilmannOSG** [comment]
Looks like this one is ready to be merged? (minus conflicts that need to be resolved)

**Comment by ruben-ayrapetyan** [comment]
Looks like not all questions were answered and one `LGTM` is missing.

**Comment by tilmannOSG** [comment]
@lvidacs Can you please address the remaining issues?

**Comment by lvidacs** [comment]
Hi, the patch is rebased and updated.

**Comment by egavrin** [comment]
Still good to me.

**Comment by egavrin** [comment]
@galpeter can we merge this?

**Comment by galpeter** [comment]
lgtm

**Comment by galpeter** [comment]
merged: https://github.com/Samsung/jerryscript/commit/cd7cf53ecb43f1072032bc1764433139258eeab8

**Pull request closed without merging**